tests/unit/components/playback/controls/PlateBarcode.spec.js

```diff
@@ -56,102 +57,50 @@ describe("PlateBarcode.vue", () => {
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 10] in which like tab is entered at [2-3] fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA  300000"); // year having special charcter
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 10] and has value such as scientific charcter degrees is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA20001º21"); // having special chacters in the last
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 10] value such as unicode charcter in hindi न is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA20210न21"); // having special unicodde chacters in the last
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 11] value such as unicode charcter in Chinese 浩 is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA20011浩211"); // having special unicode chacters in the last
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
-  test("Given a barcode with invlaid date in which the [input.length = 10] values such as unicode charcter in Chinese numbers 二千万一千〇九 (20001009) is entered  fails the defined criteria hence the Vuex barcode is blank", async () => {
-    wrapper.find("input").setValue("MA二千万一千〇九"); // having special unicode chacters of chinexe numbers 二千万一千〇九 aka 20001009
-    await wrapper.vm.$nextTick(); // wait for update
-    // confirm pre-condition
-    expect(store.state.playback.barcode).toBeNull();
-  });
+  test.each([
+    [
+      "AB200440012",
+      "validate_plate_barcode",
+      null, //  vuex value
+    ],
+    ["12200440012", "validate_plate_barcode", null],
+    ["*#200440012", "validate_plate_barcode", null],
+    ["MA200000012", "validate_plate_barcode", null],
+    ["MA203670012", "validate_plate_barcode", null],
+    ["MA209990121", "validate_plate_barcode", null],
+    ["MA 13000012", "validate_plate_barcode", null],
+    ["MA  300000", "validate_plate_barcode", null],
+    ["MB190440991", "validate_plate_barcode", "MB190440991"], // year 19 now allowed.
+    ["MB210440991", "validate_plate_barcode", "MB210440991"], // year 21 now allowed
```

Comment:
In general, the comprehensive validation of all these different possibilities should be within the `TextValidation` test file.  The goal of having it separated is that the validation function can be used by other components that trust that it should work.  Components that use the function (like PlateBarcode.vue) just need to test ~2 different right and wrong values, and also that the correct TextValidation function is being called. That is sufficient to confirm things are wired up correctly.  Each component that uses one of our javascript reusable functions shouldn't need to test it in this level of detail----but we do still need all those tests, so they should live in TextValidation.spec

Reply:
Good point, in the above retained just prevent accidental renaming of in future of validation file `TextValidation` or `rule name change` and missing out in Component.